src/main/scala/com/exasol/cloudetl/kafka/KafkaTopicDataImporter.scala

```diff
@@ -43,17 +42,19 @@ object KafkaTopicDataImporter extends LazyLogging {
     val topic = kafkaProperties.getTopic()
     val topicPartition = new TopicPartition(topic, partitionId)
 
-    val recordDeserialization = kafkaProperties.getRecordFormat() match {
-      case "avro" => AvroDeserialization
-      case "json" => JsonDeserialization
-    }
-    val recordFields = kafkaProperties.getRecordFields()
-    val derserializer = if (kafkaProperties.getSingleColJson()) {
-      recordDeserialization.getSingleColumnJsonDeserializer(kafkaProperties, recordFields)
-    } else {
-      recordDeserialization.getColumnDeserializer(kafkaProperties, recordFields)
-    }
-    val kafkaConsumer = KafkaConsumerFactory(kafkaProperties, derserializer, metadata)
+    val fieldSpecs = FieldParser.get(kafkaProperties.getRecordFields())
+
+    val recordDerse = DeserializationFactory
```

Comment:
Not sure if I'm missing some Scala convention here, not sure if Derse is a thing -- I'd expect to see Serde as a wrapper term for serializing & deserializing, though :smile: But in this case, we only seem to have deserializers, right?
```suggestion
    val recordDeserializers = DeserializationFactory
```

Reply:
I am not an scala exper either but derse does not make much sense 😃